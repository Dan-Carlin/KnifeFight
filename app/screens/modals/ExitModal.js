/*
ExitModal - Alert modal that activates when user exits the game.
*/

// External libraries
import React from "react";
import { View } from "react-native";

// Assets
import AlertIcon from "../../assets/kf_alert.svg";
import { CancelGraphic, YesGraphic } from "../../assets/buttons/actions";

// Components
import Button from "../../components/Button";
import Text from "../../components/Text";

// Resources
import strings from "../../config/strings";
import styles from "./ExitModalStyles";

function ExitModal({ onConfirm, onCancel }) {
  return (
    <View style={styles.alertContainer}>
      <View style={styles.container}>
        <View style={styles.titleContainer}>
          <View style={styles.alertIcon}>
            <AlertIcon />
          </View>
          <Text style={styles.title}>{strings.modal_exit_title}</Text>
        </View>
        <View style={styles.bodyContainer}>
          <Text style={styles.bodyText}>{strings.modal_exit_text}</Text>
        </View>
        <View style={styles.buttonsContainer}>
          <View style={styles.button}>
            <Button Graphic={CancelGraphic} onPress={onCancel} />
          </View>
          <View style={styles.button}>
            <Button Graphic={YesGraphic} onPress={onConfirm} />
          </View>
        </View>
      </View>
    </View>
  );
}

export default ExitModal;
