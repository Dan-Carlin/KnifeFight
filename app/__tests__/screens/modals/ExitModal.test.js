import React from "react";
import renderer from "react-test-renderer";

import ExitModal from "../../../screens/modals/ExitModal";

describe("Exit Modal screen tests", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<ExitModal />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
