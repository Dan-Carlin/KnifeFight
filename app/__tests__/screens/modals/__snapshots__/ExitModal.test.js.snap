// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Exit Modal screen tests renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "#414042",
        "borderColor": "#000",
        "borderRadius": 20,
        "borderWidth": 2,
        "height": "50%",
        "overflow": "hidden",
        "width": "60%",
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#0005",
          "borderBottomColor": "#000",
          "borderEndColor": "#000",
          "borderStartColor": "#bcbec0",
          "borderTopColor": "#bcbec0",
          "borderTopEndRadius": 20,
          "borderTopStartRadius": 20,
          "borderWidth": 2,
          "flex": 0.25,
          "flexDirection": "row",
          "padding": 4,
        }
      }
    >
      <View
        style={
          Object {
            "height": "100%",
            "marginEnd": 6,
            "width": "10%",
          }
        }
      >
        <SvgMock />
      </View>
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "width": "100%",
            },
            Object {
              "fontFamily": "default",
              "fontSize": 50,
            },
          ]
        }
      >
        Hold up!
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.4,
          "padding": 10,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "Avenir",
              "fontSize": 16,
              "width": "100%",
            },
            Object {
              "color": "#bcbec0",
            },
          ]
        }
      >
        Are you sure you want to end the game? 
(All data will be lost.)
      </Text>
    </View>
    <View
      style={
        Object {
          "flex": 0.35,
          "flexDirection": "row",
          "justifyContent": "flex-end",
        }
      }
    >
      <View
        style={
          Object {
            "height": "80%",
            "width": "30%",
          }
        }
      >
        <SvgMock
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        />
      </View>
      <View
        style={
          Object {
            "height": "80%",
            "width": "30%",
          }
        }
      >
        <SvgMock
          accessible={true}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
        />
      </View>
    </View>
  </View>
</View>
`;
